paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -96,50 +111,76 @@ def main() -> None:
     secret_provider_name = system_paasta_config.get_secret_provider_name()
     vault_cluster_config = system_paasta_config.get_vault_cluster_config()
     kube_client = KubeClient()
+    services_to_k8s_namespaces = get_services_to_k8s_namespaces(
+        service_list=args.service_list, cluster=cluster, soa_dir=args.soa_dir,
+    )
+
     sys.exit(0) if sync_all_secrets(
         kube_client=kube_client,
         cluster=cluster,
-        service_list=args.service_list,
+        services_to_k8s_namespaces=services_to_k8s_namespaces,
         secret_provider_name=secret_provider_name,
         vault_cluster_config=vault_cluster_config,
         soa_dir=args.soa_dir,
-        namespace=args.namespace,
+        overwrite_namespace=args.namespace,
     ) else sys.exit(1)
 
 
+def get_services_to_k8s_namespaces(
+    service_list: List[str], cluster: str, soa_dir: str,
+) -> Mapping[str, Set]:
```

Comment:
Here and everywhere below.
Isn't it `Mapping[str, Set[str]]`?

Reply:
yeah true, I forgot that typing.Set is a generic since I normally don't use sets much